homeassistant/components/history.py

```diff
@@ -185,8 +216,77 @@ def get(self, request, datetime=None):
         end_time = start_time + one_day
         entity_id = request.args.get('filter_entity_id')
 
-        return self.json(
-            get_significant_states(start_time, end_time, entity_id).values())
+        return self.json(get_significant_states(
+            start_time, end_time, entity_id, Filters(self.config)).values())
+
+
+# pylint: disable=too-few-public-methods
+class Filters(object):
+    """Container for the configured include and exclude filters."""
+
+    def __init__(self, config):
+        """Initialise the include and exclude filters."""
+        self.excluded_entities = []
+        self.excluded_domains = []
+        self.included_entities = []
+        self.included_domains = []
+        exclude = config[DOMAIN].get(CONF_EXCLUDE)
+        if exclude:
+            self.excluded_entities = exclude[CONF_ENTITIES]
+            self.excluded_domains = exclude[CONF_DOMAINS]
+        include = config[DOMAIN].get(CONF_INCLUDE)
+        if include:
+            self.included_entities = include[CONF_ENTITIES]
+            self.included_domains = include[CONF_DOMAINS]
+
+
+def _set_filters_in_query(query, states, entity_ids, filters):
```

Comment:
Also, passing in `states` is confusing (esp the name, since we also have lists/generators of states). I would prefer to use `states = recorder.get_model('States')` inside the method.